src/protocol/sort/generate_permutation.rs

```diff
@@ -158,7 +158,7 @@ where
 
     let mut composed_less_significant_bits_permutation = bit_0_permutation;
     for bit_num in 1..num_bits {
-        let ctx_bit = ctx.narrow(&Sort(bit_num));
+        let ctx_bit = ctx.narrow(&Sort(bit_num.try_into().unwrap()));
```

Comment:
We shouldn't need this casting now, right? `Sort()` accepts a usize now.  Isn't that what the type of `bit_num` is?

Reply:
This is the original generate_permutation without any optimization. It somehow works best with u32's in the code. Since we will be eliminating this code anyways, so didnt bother changing it here